examples/yaml_test.go

```diff
@@ -0,0 +1,241 @@
+)
+
+// cmd will run the command c with args and if input is provided, that will be piped
+// into the process as input
+func cmd(t *testing.T, c string, args []string, input string) (string, error) {
+	t.Helper()
+	binary, err := exec.LookPath(c)
+	if err != nil {
+		t.Fatalf("couldn't find %s, plz install", c)
+	}
+
+	t.Logf("Executing %s %v", binary, args)
+
+	cmd := exec.Command(binary, args...)
+	cmd.Env = os.Environ()
+
+	if input != "" {
+		cmd.Stdin = strings.NewReader(input)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		t.Fatalf("couldn't run command %s %v: %v, %s", c, args, err, stderr.String())
+		return "", err
+	}
+
+	return stdout.String(), nil
+}
+
+// resetState will erase all types that the yaml examples could have concievably created.
+func resetState(t *testing.T) {
+	t.Helper()
+
+	crdTypes := []string{
+		"conditions.tekton.dev",
+		"pipelineresources.tekton.dev",
+		"tasks.tekton.dev",
+		"clustertasks.tekton.dev",
+		"pipelines.tekton.dev",
+		"taskruns.tekton.dev",
+		"pipelineruns.tekton.dev",
+		"services",
+		"pods",
+		"configmaps",
+		"secrets",
+		"serviceaccounts",
+		"persistentvolumeclaims",
+	}
+	for _, c := range crdTypes {
+		cmd(t, "kubectl", []string{"delete", "--ignore-not-found=true", c, "--all"}, "")
+	}
+}
+
+// getYamls will look in the directory in examples indicated by version and run for yaml files
+func getYamls(t *testing.T, version, run string) []string {
+	t.Helper()
+	_, filename, _, _ := runtime.Caller(0)
+
+	// Don't read recursively; the only dir within these dirs is no-ci which doesn't
+	// want any tests run against it
+	files, err := ioutil.ReadDir(path.Join(path.Dir(filename), version, run))
+	if err != nil {
+		t.Fatalf("Couldn't read yaml files from %s/%s/%s: %v", path.Dir(filename), version, run, err)
+	}
+	yamls := []string{}
+	for _, f := range files {
+		if matches, _ := filepath.Match("*.yaml", f.Name()); matches {
+			yamls = append(yamls, f.Name())
+		}
+	}
+	return yamls
+}
+
+// replaceDockerRepo will look in the content f and replace the hard coded docker
+// repo with the on provided via the KO_DOCKER_REPO environment variable
+func replaceDockerRepo(t *testing.T, f string) string {
+	t.Helper()
+	r := os.Getenv("KO_DOCKER_REPO")
+	if r == "" {
+		t.Fatalf("KO_DOCKER_REPO must be set")
+	}
+	read, err := ioutil.ReadFile(f)
+	if err != nil {
+		t.Fatalf("couldnt read contents of %s: %v", f, err)
+	}
+	return strings.Replace(string(read), "gcr.io/christiewilson-catfactory", r, -1)
+}
+
+// logRun will retrieve the entire yaml of run and log it
+func logRun(t *testing.T, run string) {
+	t.Helper()
+	yaml, err := cmd(t, "kubectl", []string{"get", run, "-o", "yaml"}, "")
+	if err == nil {
+		t.Logf(yaml)
+	}
+}
+
+// pollRun will use kubectl to query the specified run to see if it
+// has completed. It will timeout after timeoutSeconds.
+func pollRun(t *testing.T, run string, wg *sync.WaitGroup) {
```

Comment:
This could Watch instead and maybe end up faster?

Reply:
hm could you give me an example of how that would work? ive never used kubectl watch - trying it out now it seems like id have to stream output from it which seems a bit more complicated given im calling kubectl with exec.Command - i think im being a bit lazy for sure but this seems pretty much fine for now? i can put in a comment to explore using watch